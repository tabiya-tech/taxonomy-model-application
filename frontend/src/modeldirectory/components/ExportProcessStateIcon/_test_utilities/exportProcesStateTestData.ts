import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ExportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllExportProcessStatePermutations = (): ModelInfoTypes.ExportProcessState[] => {
  // generate all permutations of ExportProcessState status, errored, exportErrors, exportWarnings
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ExportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((exportErrors) => {
        allBooleans.forEach((exportWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              exportErrors: exportErrors,
              exportWarnings: exportWarnings,
            },
            downloadUrl: "https://example.com",
            timestamp: new Date(),
          });
        });
      });
    });
  });
  return allPermutations;
};
