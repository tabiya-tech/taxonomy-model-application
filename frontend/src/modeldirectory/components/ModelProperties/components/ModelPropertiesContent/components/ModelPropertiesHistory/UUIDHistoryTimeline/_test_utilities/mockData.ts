import { ModelInfoTypes } from "src//modelInfo/modelInfoTypes";

export function getOneFakeUUIDHistoryDetail(i: number): ModelInfoTypes.ModelHistory {
  return {
    UUID: "e26aca13-aa00-4346-9e83-edaf59314378" + i,
    id: "5e12f537f079fbdcb4e61bf" + i,
    localeShortCode: "ZA",
    name: "Model" + i,
    version: "V1.1." + i,
  };
}

export function getFakeUUIDHistoryDetailsArray(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => getOneFakeUUIDHistoryDetail(i));
}

export function getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => {
    const details = getOneFakeUUIDHistoryDetail(i);
    details.id = null;
    details.name = null;
    details.version = null;
    details.localeShortCode = null;
    return details;
  });
}
