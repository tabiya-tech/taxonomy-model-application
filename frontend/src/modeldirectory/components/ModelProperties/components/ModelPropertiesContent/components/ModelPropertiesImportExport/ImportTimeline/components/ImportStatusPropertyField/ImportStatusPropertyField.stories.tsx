import { Meta, StoryObj } from "@storybook/react";
import ImportStatusPropertyField, { uniqueId } from "./ImportStatusPropertyField";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";
import { v4 as id } from "uuid";

const meta: Meta<typeof ImportStatusPropertyField> = {
  title: "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesImportExport/ImportTimeline/ImportStatusPropertyField",
  component: ImportStatusPropertyField,
  tags: ["autodocs"],
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.PENDING,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  },
};

export default meta;

type Story = StoryObj<typeof ImportStatusPropertyField>;

export const Pending: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.PENDING,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const Running: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.RUNNING,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const Completed: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const CompletedWithCriticalErrors: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: true,
        parsingWarnings: false,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}


export const CompletedWithParsingErrors: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: false,
        parsingErrors: true
      }
    },
    fieldId: uniqueId
  }
}

export const CompletedWithParsingWarnings: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: true,
        parsingErrors: false
      }
    },
    fieldId: uniqueId
  }
}

export const CompletedWithParsingWarningsAndParsingErrors: Story = {
  args: {
    importProcessState: {
      id: id(),
      status: ImportProcessStateEnums.Status.COMPLETED,
      result: {
        errored: false,
        parsingWarnings: true,
        parsingErrors: true
      }
    },
    fieldId: uniqueId
  }
}
