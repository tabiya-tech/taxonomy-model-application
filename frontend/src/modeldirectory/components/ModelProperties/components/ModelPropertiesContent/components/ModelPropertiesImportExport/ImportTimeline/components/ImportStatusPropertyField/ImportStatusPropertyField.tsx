import React from "react";
import Box from "@mui/material/Box";
import HelpTip from "src/theme/HelpTip/HelpTip";
import { Typography, useTheme } from "@mui/material";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ImportProcessStateEnums from "api-specifications/importProcessState";
import ImportProcessStateIcon from "src/modeldirectory/components/ImportProcessStateIcon/ImportProcessStateIcon";
import PropertyFieldLayout from "src/theme/PropertyFieldLayout/PropertyFieldLayout";

interface ImportStatusPropertyFieldProps {
  importProcessState: ModelInfoTypes.ImportProcessState;
  fieldId: string;
}

export const uniqueId = "56ea04a0-f80e-4cd0-b967-eb55d108f1b6";

export const DATA_TEST_ID = {
  IMPORT_STATUS_FIELD: `status-${uniqueId}`,
  PROCCESS_STATUS_ICON: `process-status-icon-${uniqueId}`,
  MESSAGE: `message-${uniqueId}`
};

/**
 * ImportStatusPropertyField is responsible for showing the import status
 * @param props
 * @constructor
 */

const ImportStatusPropertyField: React.FC<ImportStatusPropertyFieldProps> = (
  props: Readonly<ImportStatusPropertyFieldProps>
) => {
  const theme = useTheme();

  const Meessage = () => {
    const { status, result } = props.importProcessState

    if(status === ImportProcessStateEnums.Enums.Status.PENDING) {
      return <Typography data-testid={DATA_TEST_ID.MESSAGE}>Pending</Typography>
    }

    if(status === ImportProcessStateEnums.Enums.Status.RUNNING) {
      return <Typography data-testid={DATA_TEST_ID.MESSAGE}>Running</Typography>
    }

    if((status === ImportProcessStateEnums.Enums.Status.COMPLETED) && !result.errored && !result.parsingErrors && !result.parsingWarnings) {
      return <Typography data-testid={DATA_TEST_ID.MESSAGE}>Completed successfully</Typography>
    }

    if((status === ImportProcessStateEnums.Enums.Status.COMPLETED) && result.errored && !result.parsingErrors && !result.parsingWarnings) {
      return (
          <Typography data-testid={DATA_TEST_ID.MESSAGE}>
            Completed with
            <Box component={"span"} fontWeight={"bold"}> critical errors</Box>
            <HelpTip>The import process could not start or complete due to a technical error. Please try again later or contact us if the problem persists.</HelpTip>
          </Typography>
      )
    }

    if((status === ImportProcessStateEnums.Enums.Status.COMPLETED) && !result.errored && result.parsingErrors && !result.parsingWarnings) {
      return (
          <Typography data-testid={DATA_TEST_ID.MESSAGE}>
            Completed with
            <Box component={"span"} fontWeight={"bold"}> parsing errors</Box>
            <HelpTip>
              The esv files have inconsistencies that could not be resolved. The model was created but is not consistent.
              Please review the csv files and try again.
            </HelpTip>
          </Typography>
      )
    }

    if((status === ImportProcessStateEnums.Enums.Status.COMPLETED) && !result.errored && !result.parsingErrors && result.parsingWarnings) {
      return (
        <Typography data-testid={DATA_TEST_ID.MESSAGE}>
          Completed with
          <Box component={"span"} fontWeight={"bold"}> parsing warnings</Box>
          <HelpTip>
            One or more rows or some field could not be imported, but the model was still created and is consistent.
            Please review the csv files and try again.
          </HelpTip>
        </Typography>
      )
    }

    if((status === ImportProcessStateEnums.Enums.Status.COMPLETED) && !result.errored && result.parsingErrors && result.parsingWarnings) {
      return (
        <Typography data-testid={DATA_TEST_ID.MESSAGE}>
          <Box component={"span"}>Completed with </Box>
          <Box component={"span"} fontWeight={"bold"}>
            parsing errors
          </Box>
          <HelpTip>
            The esv files have inconsistencies that could not be resolved. The model was created but is not consistent.
            Please review the csv files and try again.
          </HelpTip>
          and
          <Box component={"span"} fontWeight={"bold"}> parsing warnings</Box>
          <HelpTip>
            One or more rows or some field could not be imported, but the model was still created and is consistent.
            Please review the csv files and try again.
          </HelpTip>
        </Typography>
      )
    }

    return <Typography>Unknown Status</Typography>
  }

  return (
    <PropertyFieldLayout title={"Status"} fieldId={props.fieldId}>
      <Box display={"flex"} gap={theme.spacing(1)} alignItems={"center"} data-testid={DATA_TEST_ID.IMPORT_STATUS_FIELD}>
        <ImportProcessStateIcon test-dataId={DATA_TEST_ID.PROCCESS_STATUS_ICON} importProcessState={props.importProcessState} />
        <Meessage/>
      </Box>
    </PropertyFieldLayout>
  )
};

export default ImportStatusPropertyField;
