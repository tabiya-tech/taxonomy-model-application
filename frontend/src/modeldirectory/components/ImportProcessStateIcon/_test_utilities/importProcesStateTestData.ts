import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllImportProcessStatePermutations = (): ModelInfoTypes.ImportProcessState[] => {
  // generate all permutations of ImportProcessState status, errored, parsingErrors, parsingWarnings
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ImportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((parsingErrors) => {
        allBooleans.forEach((parsingWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              parsingErrors: parsingErrors,
              parsingWarnings: parsingWarnings,
            },
            createdAt: new Date(allPermutations.length),
            updatedAt: new Date(allPermutations.length),
          });
        });
      });
    });
  });
  return allPermutations;
};
