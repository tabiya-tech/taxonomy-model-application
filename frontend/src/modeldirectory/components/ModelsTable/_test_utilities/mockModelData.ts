import ModelInfoAPISpecs from "api-specifications/modelInfo";
import LocaleAPISpecs from "api-specifications/locale";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import ExportProcessStateAPISpecs from "api-specifications/exportProcessState";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";

import { v4 as uuidv4 } from "uuid";

import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import { getMockId } from "src/_test_utilities/mockMongoId";
import { getRandomLorem, getRandomString, getTestString } from "src/_test_utilities/specialCharacters";
import { faker } from "@faker-js/faker";
import { CELL_MAX_LENGTH } from "src/modeldirectory/components/ModelsTable/ModelsTable";

export function getOneFakeModel(id?: number): ModelInfoTypes.ModelInfo {
  const model = getArrayOfFakeModels(1)[0];
  model.id = getMockId(id ?? 1);
  return model;
}

export function getArrayOfFakeModels(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: faker.location
          .countryCode("alpha-3")
          .substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description:
        i % 2 === 0
          ? getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH)
          : getRandomLorem(CELL_MAX_LENGTH / 2), // 50% chance of long description
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.system.semver().substring(0, ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getArrayOfFakeModelsMaxLength(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: getRandomLorem(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getRandomLorem(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: getRandomLorem(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getRandomLorem(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getOneRandomModelMaxLength(): ModelInfoTypes.ModelInfo {
  return getArrayOfRandomModelsMaxLength(1)[0];
}

export function getArrayOfRandomModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: getRandomString(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getTestString(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0,
      releaseNotes: getTestString(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export const fakeModel: ModelInfoTypes.ModelInfo = {
  UUID: "8d914eab-6f7d-4183-accc-a09f99887b39",
  UUIDHistory: ["8d914eab-6f7d-4183-accc-a09f99887b39"],
  description: "aw j zt   agrkasl dy ogtimpsauwumu l utrovthao syertm beawpxluhyudgzbbm",
  id: "000000000000000000000001",
  importProcessState: {
    id: "000000000000000000000001",
    result: {
      errored: true,
      parsingErrors: true,
      parsingWarnings: true,
    },
    status: ImportProcessStateEnums.Status.PENDING,
    createdAt: new Date("2023-10-18T17:35:10.571Z"),
    updatedAt: new Date("2023-10-18T17:35:12.571Z"),
  },
  exportProcessState: [
    {
      downloadUrl: "https://www.example.com/",
      id: "000000000000000000000001",
      result: {
        errored: true,
        exportErrors: true,
        exportWarnings: true,
      },
      status: ExportProcessStateAPISpecs.Enums.Status.PENDING,
      timestamp: new Date("2023-10-18T17:35:10.571Z"),
      createdAt: new Date("2023-10-18T17:35:10.571Z"),
      updatedAt: new Date("2023-10-18T17:35:12.571Z"),
    },
  ],
  locale: {
    UUID: "7c4ea00b-fbea-4a31-9c04-009c813f5f34",
    name: "Palestine",
    shortCode: "ETH",
  },
  name: "1/1 - fysmros tcgcnjbbrrev",
  path: "https://unwelcome-editorial.net/",
  releaseNotes: "Perferendis modi impedit necessitatibus a",
  released: true,
  tabiyaPath: "https://silver-vulture.biz",
  createdAt: new Date("2023-10-18T17:35:10.571Z"),
  updatedAt: new Date("2022-10-18T17:35:10.571Z"),
  version: "0.2.2",
};

export function getRandomImportStatus(id: number) {
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getRandomExportStatus(id: number) {
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getOneFakeExportProcessState(i: number): ModelInfoTypes.ExportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomExportStatus(i),
    result: {
      errored: i % 2 === 0,
      exportErrors: i % 2 === 0,
      exportWarnings: i % 2 === 0,
    },
    downloadUrl: faker.internet.url(),
    timestamp: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}

export function getOneFakeImportProcessState(i: number): ModelInfoTypes.ImportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomImportStatus(i),
    result: {
      errored: i % 2 === 0,
      parsingErrors: i % 2 === 0,
      parsingWarnings: i % 2 === 0,
    },
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}
