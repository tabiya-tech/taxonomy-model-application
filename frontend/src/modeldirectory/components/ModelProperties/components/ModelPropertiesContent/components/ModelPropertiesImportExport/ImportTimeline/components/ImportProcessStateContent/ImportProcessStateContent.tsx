import { useTheme } from "@mui/material";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import React from "react";
import Box from "@mui/material/Box";
import ImportStatusPropertyField from "../ImportStatusPropertyField/ImportStatusPropertyField";
import DurationPropertyField from "src/theme/PropertyFieldLayout/DurationPropertyField/DurationPropertyField";

interface ImportProcessStateContentProps {
  importProcessState: ModelInfoTypes.ImportProcessState;
}

const uniqueId = "333ab3d5-61a7-4169-8af9-01fe57ed5673";

export const DATA_TEST_ID = {
  IMPORT_PROCESS_STATE_CONTENT: `import-process-state-content-${uniqueId}`,
  IMPORT_STATUS_FIELD: `import-status-${uniqueId}`,
  IMPORT_DURATION_FIELD: `import-duration-${uniqueId}`,
};

export const FIELD_ID = {
  IMPORT_STATUS_FIELD: `import-status-${uniqueId}`,
  IMPORT_DURATION_FIELD: `import-duration-${uniqueId}`,
};

/**
 * ImportProcessStateContent is responsible for showing the import status and duration
 * @param props
 * @constructor
 */

const ImportProcessStateContent: React.FC<ImportProcessStateContentProps> = (
  props: Readonly<ImportProcessStateContentProps>
) => {
  const theme = useTheme();

  return (
    <Box
      display="flex"
      flexDirection="column"
      gap={theme.tabiyaSpacing.md}
      paddingY={theme.tabiyaSpacing.lg}
      data-testid={DATA_TEST_ID.IMPORT_PROCESS_STATE_CONTENT}
    >
      <ImportStatusPropertyField
        importProcessState={props.importProcessState}
        fieldId={FIELD_ID.IMPORT_STATUS_FIELD}
        data-testid={DATA_TEST_ID.IMPORT_STATUS_FIELD}
      />
      <DurationPropertyField
        label="Duration"
        firstDate={props.importProcessState.createdAt ? props.importProcessState.createdAt : new Date()}
        secondDate={props.importProcessState.updatedAt}
        fieldId={FIELD_ID.IMPORT_DURATION_FIELD}
        data-testid={DATA_TEST_ID.IMPORT_DURATION_FIELD}
      />
    </Box>
  );
};

export default ImportProcessStateContent;
