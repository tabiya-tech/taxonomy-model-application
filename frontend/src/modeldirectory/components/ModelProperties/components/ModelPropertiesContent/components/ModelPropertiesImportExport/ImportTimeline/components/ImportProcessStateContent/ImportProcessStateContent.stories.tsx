import { Meta, StoryObj } from "@storybook/react";
import { getOneFakeImportProcessState } from "src/modeldirectory/components/ModelsTable/_test_utilities/mockModelData";
import ImportProcessStateContent from "./ImportProcessStateContent";

const meta: Meta<typeof ImportProcessStateContent> = {
  title:
    "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesImportExportComponents/ImportProcessStateContent",
  component: ImportProcessStateContent,
  tags: ["autodocs"],
  args: {
    importProcessState: getOneFakeImportProcessState(1),
  },
};

type Story = StoryObj<typeof ImportProcessStateContent>;

export const Shown: Story = {};

export default meta;
