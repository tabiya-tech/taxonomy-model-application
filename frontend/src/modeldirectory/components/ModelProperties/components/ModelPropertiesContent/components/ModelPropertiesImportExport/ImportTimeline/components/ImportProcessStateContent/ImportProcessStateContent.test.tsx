// mute the console
import "src/_test_utilities/consoleMock";

import { render, screen } from "src/_test_utilities/test-utils";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";
import ImportProcessStateContent, { FIELD_ID, DATA_TEST_ID } from "./ImportProcessStateContent";
import ImportStatusPropertyField from "../ImportStatusPropertyField/ImportStatusPropertyField";
import DurationPropertyField from "src/theme/PropertyFieldLayout/DurationPropertyField/DurationPropertyField";

// mock the ImportStatusPropertyField component
jest.mock(
  "src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField",
  () => {
    return {
      __esModule: true,
      default: jest.fn().mockImplementation((props) => (
        <div data-testid={props["data-testid"]} id={props.fieldId}>
          ImportStatusPropertyField Mock
        </div>
      )),
    };
  }
);

// mock the DurationPropertyField component
jest.mock("src/theme/PropertyFieldLayout/DurationPropertyField/DurationPropertyField", () => {
  return {
    __esModule: true,
    default: jest.fn().mockImplementation((props) => (
      <div data-testid={props["data-testid"]} id={props.fieldId}>
        Duration Property Field Mock
      </div>
    )),
  };
});

describe("ImportProcessStateContent", () => {
  test("should render correctly with the provided props", () => {
    // GIVEN an import process state
    const givenImportProcessState = {
      id: "000000000000000000000001",
      result: {
        errored: true,
        parsingErrors: true,
        parsingWarnings: true,
      },
      status: ImportProcessStateEnums.Status.PENDING,
      createdAt: new Date("Wed, Mar 6, 2024, 11:02 AM"),
    };

    // WHEN the component is rendered
    render(<ImportProcessStateContent importProcessState={givenImportProcessState} />);

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the component to be shown
    const actualContentContainer = screen.getByTestId(DATA_TEST_ID.IMPORT_PROCESS_STATE_CONTENT);
    expect(actualContentContainer).toBeInTheDocument();
    // AND the ImportStatusPropertyField component to be shown
    const actualImportStatusField = screen.getByTestId(DATA_TEST_ID.IMPORT_STATUS_FIELD);
    expect(actualImportStatusField).toBeInTheDocument();
    // AND the ImportStatusPropertyField to be called with the correct props
    expect(ImportStatusPropertyField).toHaveBeenCalledWith(
      {
        importProcessState: givenImportProcessState,
        fieldId: FIELD_ID.IMPORT_STATUS_FIELD,
        "data-testid": DATA_TEST_ID.IMPORT_STATUS_FIELD,
      },
      {}
    );
    // AND the DurationPropertyField component to be shown
    const actualDurationField = screen.getByTestId(DATA_TEST_ID.IMPORT_DURATION_FIELD);
    expect(actualDurationField).toBeInTheDocument();
    // AND the DurationPropertyField to be called with the correct props
    expect(DurationPropertyField).toHaveBeenCalledWith(
      {
        label: "Duration",
        firstDate: givenImportProcessState.createdAt,
        secondDate: undefined,
        fieldId: FIELD_ID.IMPORT_DURATION_FIELD,
        "data-testid": DATA_TEST_ID.IMPORT_DURATION_FIELD,
      },
      {}
    );
    // AND to match the snapshot
    expect(actualContentContainer).toMatchSnapshot();
  });
});
