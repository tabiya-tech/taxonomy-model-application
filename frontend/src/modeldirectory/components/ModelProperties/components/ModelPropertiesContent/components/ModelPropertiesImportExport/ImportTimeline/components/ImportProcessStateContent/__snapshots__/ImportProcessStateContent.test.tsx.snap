// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportProcessStateContent should render correctly with the provided props 1`] = `
<div
  class="MuiBox-root css-2rg0ju"
  data-testid="import-process-state-content-333ab3d5-61a7-4169-8af9-01fe57ed5673"
>
  <div
    data-testid="import-status-333ab3d5-61a7-4169-8af9-01fe57ed5673"
    id="import-status-333ab3d5-61a7-4169-8af9-01fe57ed5673"
  >
    ImportStatusPropertyField Mock
  </div>
  <div
    data-testid="import-duration-333ab3d5-61a7-4169-8af9-01fe57ed5673"
    id="import-duration-333ab3d5-61a7-4169-8af9-01fe57ed5673"
  >
    Duration Property Field Mock
  </div>
</div>
`;
